modules/virt-template-fields-for-boot-source.adoc

```diff
@@ -0,0 +1,53 @@
+// Module included in the following assemblies:
+//
+// * virt/vm_templates/virt-creating-vm-template.adoc
+
+[id="virt-template-fields-for-boot-source_{context}"]
+= Virtual machine template fields for adding a boot source
+
+The following table describes the fields for *Add boot source to template* dialog. This dialog displays when you click *Add Source* for a virtual machine template in the *Templates* tab.
+[cols="2a,3a,5a"]
+|===
+|Name | Parameter |  Description
+
+.7+|Boot source type
+|Blank (creates PVC)
+|Create an empty disk. This option displays if you use the interactive wizard to add a disk.
+
+|Upload local file (creates PVC)
+|Drag a file or browse to upload it from your filesystem.
+
+|Import via URL (creates PVC)
+|Obtain the download link URL from the web page where the image download is available and enter that URL link in the *Import URL* field. Example: For a Red Hat Enterprise Linux image, log on to the Red Hat customer portal, access the image download page, and copy the download link URL for the KVM guest image.
+
+|Use an existing PVC
+|Use a PVC that is already available in the cluster and clone it. This option displays if you use the interactive wizard to add a disk.
+
+|Clone existing PVC (creates PVC)
+|Select a Persistent Volume Claim (PVC) from a list of available PVCs.
+
+|Container image (creates PVC)
+|Specify the bootable operating system container that is located in a registry and accessible from the cluster. Example: kubevirt/cirros-registry-dis-demo.
+
+|Container (ephemeral)
+|Upload content from a container located in a registry that is accessible from the cluster. Use the container disk for read-only filesystems such as CD-ROMs or temporary virtual machines. This option displays if you use the interactive wizard to create a disk.
+
+|Source provider
+|
+|Optional field. Add descriptive text about the source for the template or the name of the user who created the template. Example: Red Hat.
+
+.3+|Advanced
+|Storage class
+|The storage class that is used to create the disk.
+
+|Access mode
+|Access mode of the persistent volume. Supported access modes are *Single User (RWO)*, *Shared Access (RWX)*, *Read Only (ROX)*. If *Single User (RWO)* is selected, the disk can be mounted as read/write by a single node. If *Single User (RWO)* is selected, the disk can be mounted as read-only by many nodes.
+[NOTE]
+====
+Shared Access (RWX) is required for some features, such as live migration of virtual machines between nodes.
+====
+
+|Volume mode
+|Defines whether the persistent volume uses a formatted file system or raw block state. Supported modes are *Block* and *Filesystem*. Default is *Filesystem*.
```

Comment:
Default value is set according to the best options available for storage class selected, the mapping is done using a config map named "kubevirt-storage-class-defaults" that is created during kubevirt install and should store the best options for each storage class installed in the cluster.

Reply:
Updated the doc content, please review.